src/main/java/com/minecolonies/coremod/generation/SifterCraftingProvider.java

```diff
@@ -0,0 +1,364 @@
+            this.lootTable = lootTable;
+        }
+
+        @NotNull
+        public String getName() { return name; }
+
+        @NotNull
+        public Item getMesh() { return mesh; }
+
+        public int getMinBuildingLevel() { return minBuildingLevel; }
+
+        @NotNull
+        public LootTableJson getLootTable() { return lootTable; }
+    }
+
+    private static class LootTableBuilder
+    {
+        private final List<EntryJson> entries = new ArrayList<>();
+        private int rolls = 1;
+        private float bonusRolls = 0f;
+
+        @NotNull
+        public LootTableBuilder rolls(final int rolls)
+        {
+            this.rolls = rolls;
+            return this;
+        }
+
+        @NotNull
+        public LootTableBuilder bonusRolls(final float bonusRolls)
+        {
+            this.bonusRolls = bonusRolls;
+            return this;
+        }
+
+        @NotNull
+        public LootTableBuilder empty(final int weight)
+        {
+            entries.add(new EntryJson(null, EntryTypeEnum.EMPTY, null, null, false, null, weight, 0));
+            return this;
+        }
+
+        @NotNull
+        public LootTableBuilder item(final Item item, final int weight)
+        {
+            entries.add(new EntryJson(null, EntryTypeEnum.ITEM, item.getRegistryName().toString(), null, false, null, weight, 0));
+            return this;
+        }
+
+        @NotNull
+        public LootTableJson build()
+        {
+            return new LootTableJson(LootTableTypeEnum.GENERIC,
+                    Collections.singletonList(new PoolJson(null, rolls, bonusRolls, entries)));
+        }
+    }
+
+    private class SifterRecipeProvider extends CustomRecipeProvider
+    {
+        public SifterRecipeProvider(@NotNull final DataGenerator generatorIn)
+        {
+            super(generatorIn);
+        }
+
+        @Override
+        protected void registerRecipes(@NotNull final Consumer<IFinishedRecipe> consumer)
+        {
+            for (final Map.Entry<Item, List<SifterMeshDetails>> inputEntry : inputs.entrySet())
+            {
+                for (final SifterMeshDetails mesh : inputEntry.getValue())
+                {
+                    final String name = mesh.getName() + "/" + inputEntry.getKey().getRegistryName().getPath();
+                    final Stream<Item> loot = mesh.getLootTable().getPools().stream()
+                            .flatMap(pool -> pool.getEntries().stream())
+                            .filter(entry -> entry.getType().equals(EntryTypeEnum.ITEM))
+                            .map(entry -> ForgeRegistries.ITEMS.getValue(new ResourceLocation(entry.getName())));
+
+                    CustomRecipeBuilder.create(ModJobs.SIFTER_ID.getPath(), name)
+                            .inputs(Stream.of(new ItemStack(inputEntry.getKey()), new ItemStack(mesh.getMesh())).collect(Collectors.toList()))
+                            .secondaryOutputs(Stream.concat(Stream.of(mesh.getMesh()), loot)

```

Comment:
It may be worth adding an additional filter here for the very low probability items.

Reply:
To hide them, you mean?  I'm not sure that's needed, there's nothing below 1% probability and it is useful to know that there is at least _some_ chance of producing an item there.  Otherwise the couriers will take it away and you have no idea where it's coming from.

Reply:
Well, I guess a few things are under 1%.  But not by much.